types/cli-table2/cli-table2-tests.ts

```diff
@@ -0,0 +1,232 @@
+import * as Table from "cli-table2";
+
+// Code snippets from README.md
+
+// Horizontal table
+const table1 = new Table({ head: ['TH 1 label', 'TH 2 label'], colWidths: [100, 200] }) as Table.HorizontalTable;
+table1.push(
+    ['First value', 'Second value'],
+    ['First value', 'Second value']
+);
+
+// Vertical table
+const table2 = new Table() as Table.VerticalTable;
```

Comment:
So what makes this cast valid? Do the constructor options somehow determine it, or is the cast just an expression of intent?